include/rmm/mr/device/detail/arena.hpp

```diff
@@ -154,6 +156,26 @@ constexpr std::size_t align_down(std::size_t v) noexcept
   return rmm::detail::align_down(v, rmm::detail::CUDA_ALLOCATION_ALIGNMENT);
 }
 
+/**
+ * @brief Return human readable number of bytes.
+ *
+ * @param bytes number of bytes
+ * @return Return human readable number of bytes.
+ */
+inline std::string human_size(std::size_t bytes)
```

Comment:
I would replace this with a `formatter<bytes>` like this (and possibly put it in a more central rmm header, like `logger.hpp`). 

```
#include <fmt/format.h>

struct bytes {
  int value;
};

namespace fmt {
template <>
struct formatter<bytes>: formatter<double> {
  template <typename FormatContext>
  auto format(bytes b, FormatContext& ctx) {
    std::string const units[] = {"B", "KiB", "MiB", "GiB", "TiB"};
    int i                     = 0;
    auto size                 = static_cast<double>(b.value);
    while (size > 1024) {
      size /= 1024;
      i++;
    }
    auto out = formatter<double>::format(size, ctx);
    *out++ = units[i];
    return out;
  }
};
```

This can be used like the following, which I think is clearer.

```
logger->info("  Maximum size: {}", bytes{maximum_size_});
// or e.g. to print with fmtlib:
fmt::print("{}", bytes{1200}); // prints 1.2kB
```

Note: inspired by: https://github.com/fmtlib/fmt/issues/757

Reply:
Hmm, I'm having trouble getting this to work, I think it's because we are using the fmt headers bundled by spdlog. Adding fmt separately seems to cause some conflict with spdlog. Using the stream insertion operator seems work, and we also need it for the thread id.